docs/dev/triage-rocmlir.rst

```diff
@@ -0,0 +1,144 @@
+       %4 = memref.expand_shape %arg2 [[0, 1, 2, 3, 4]] output_shape [1, 1, 1, 1, 16] : memref<16xf32> into memref<1x1x1x1x16xf32>
+       linalg.generic {indexing_maps = [#map1, #map2, #map1], iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel"]} ins(%0, %4 : memref<1x1x30x30x16xf32>, memref<1x1x1x1x16xf32>) outs(%arg3 : memref<1x1x30x30x16xf32>) {
+       ^bb0(%arg4: f32, %arg5: f32, %arg6: f32):
+         %8 = arith.addf %arg4, %arg5 : f32
+         linalg.yield %8 : f32
+       }
+       return
+     }
+   }
+
+Please provide only the *SINGLE* module that fails and not a dump of
+them for rocMLIR team figure out which is failing.
+
+Step 3 See if its a B2 - runtime failure of MLIR-generated kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+So kernels are first run in the benchmarking process of MIGraphX.
+Therefore, set ``MIGRAPHX_TRACE_BENCHMARKING=3``
+
+You should see
+
+::
+
+   Benchmarking mlir_* : xx configs
+   Problem: <problem key>
+   Benchmarking solution: <perf config>
+
+So you might want to report that to rocMLIR team. However, if its a
+fusion specific runtime issue, further triaging is needed.
+
+You would need to manually go through the traces with
+``MIGRAPHX_TRACE_MLIR=2`` and figure out which problem has the
+``<problem key>`` that fails.
+
+Then report the failing module as in Step 2.2 to rocMLIR team. This can
+be very intensive unless some action is taken for
+https://github.com/ROCm/AMDMIGraphX/issues/2332
+
+Step 4 See if its a B3 - accuracy issue of MLIR-generated kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the hardest type of triages for rocMLIR.
+
+Here you would need to get MIGraphX modules (not MLIR) that gets printed
+in with ``MIGRAPHX_TRACE_MLIR=2``
+
+For e.g. :
+
+::
+
+   # This is MIGRAPHX module
+
+   arg2 = @param:arg2 -> float_type, {1, 5, 3}, {15, 3, 1}
+   arg1 = @param:arg1 -> float_type, {1, 4, 3}, {12, 3, 1}
+   arg0 = @param:arg0 -> float_type, {1, 5, 4}, {20, 4, 1}
+   @3 = dot(arg0,arg1) -> float_type, {1, 5, 3}, {15, 3, 1}
+   @4 = add(@3,arg2) -> float_type, {1, 5, 3}, {15, 3, 1}
+   @5 = @return(@4)
+
+   # This is the MLIR module
+
+   module {
+     func.func @mlir_dot_add(%arg0: !migraphx.shaped<1x5x4xf32, 20x4x1>, %arg1: !migraphx.shaped<1x4x3xf32, 12x3x1>, %arg2: !migraphx.shaped<1x5x3xf32, 15x3x1>) -> !migraphx.shaped<1x5x3xf32, 15x3x1> attributes {arch = "gfx90a:sramecc+:xnack-", enable_splitk_for_tuning = true, kernel = "mixr", num_cu = 110 : i64} {
+       %0 = migraphx.dot %arg0, %arg1 : <1x5x4xf32, 20x4x1>, <1x4x3xf32, 12x3x1> -> <1x5x3xf32, 15x3x1>
+       %1 = migraphx.add %0, %arg2 : <1x5x3xf32, 15x3x1>, <1x5x3xf32, 15x3x1> -> <1x5x3xf32, 15x3x1>
+       return %1 : !migraphx.shaped<1x5x3xf32, 15x3x1>
+     }
+   }
+
+Then individually create MIGraphX program that only has the MIGRAPHX
+module Then indiviually ``driver verify`` them to see which is the
+failing module.
```

Comment:
I don't think MIGraphX is able to do this easily

Reply:
yes but someone has to do it, right ?

Reply:
https://github.com/ROCm/AMDMIGraphX/pull/3182/files# 

With this PR it would be possible to dump MIGraphX programs for each module as MXR and then migraphx-driver can driver verify them.